utils/compilation_filter.py

```diff
@@ -0,0 +1,259 @@
+
+    parser = argparse.ArgumentParser(
+        description="Filter compilation and optimiztion.")
+    parser.add_argument("--db-file")
+    parser.add_argument("--filter-target")
+    options = parser.parse_args()
+
+    assert options.db_file and options.filter_target, "Please specify db file and filter target."
+    return options.db_file, options.filter_target
+
+
+def init_db(sqliteFile: str) -> sqlite3.Connection:
+    """Initialize a sqlite3 database connection."""
+
+    assert os.path.isfile(sqliteFile)
+
+    # Connect to database file.
+    return sqlite3.connect(sqliteFile)
+
+
+def find_all_related_transformation(
+        cursor: sqlite3.Cursor,
+        transIDs: List[str]):
+    """A recursive function that find all related transformations given a list of transformation IDs in the database.
+
+    Args:
+        cursor: sqlite3.Cursor. Cursor of current sqlite3 database connection.
+        transIDs: List[str]. A list of transformation IDs.
+    """
+
+    transQueryStr = "(" + ', '.join(transIDs) + ')'
+    cursor.execute(f"""
+            SELECT node_name
+            FROM Log_Transformation
+            WHERE trans_id in {transQueryStr}
+            GROUP BY node_name
+        """)
+    rows = cursor.fetchall()
+    nodesList = ["'" + r[0] + "'" for r in rows]
+
+    transQueryStr = "(" + ', '.join(nodesList) + ')'
+    cursor.execute(f"""
+            SELECT trans_id
+            FROM Log_Transformation
+            WHERE node_name in {transQueryStr}
+            GROUP BY trans_id
+        """)
+    rows = cursor.fetchall()
+    newTransIDs = [str(r[0]) for r in rows]
+
+    if sorted(newTransIDs) != sorted(transIDs):
+        transIDs = find_all_related_transformation(cursor, newTransIDs)
+    return transIDs
+
+
+def filter_node_transformation(nodeName: str, conn: sqlite3.Connection):
+    """Filter out all node transformation that is related to the given node.
+
+    Args:
+        nodeName: str. The node name that is passed to this script.
+        conn: sqlite3.Connection. A sqlite3 database connection.
+    """
+
+    cursor = conn.cursor()
+    cursor.execute("""
+            SELECT trans_id
+            FROM Log_Transformation
+            WHERE node_name = ?
+            GROUP BY trans_id
+        """, (nodeName,))
+    rows = cursor.fetchall()
+
+    directTransIDs = [str(r[0]) for r in rows]
+
+    transIDs = find_all_related_transformation(cursor, directTransIDs)
+
+    for tid in transIDs:
+        cursor.execute("""
+            SELECT *
+            FROM Log_Transformation
+            WHERE trans_id = ?
+        """, (tid, ))
+        rows = cursor.fetchall()
+        if len(rows):
+            tran = Transformation(tid)
+            if tid in directTransIDs:
+                tran.isDirectTrans_ = True
+            TRANS_LIST.append(tran)
+            tran.scopeName_ = rows[0][4].replace(
+                "glow::", "").replace(
+                "->", r" --\> ")
+            for r in rows:
```

Comment:
This would be clearer with symbolic names for the fields, e.g.

`(type, kind, name) = r[:3]`